client/FileProcessor.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include <string>
+#include <vector>
+#include <memory>
+#include <fstream>
+#include <iostream>
+#include <stdexcept>
+#include <cstdint>
+#include <cstring>
+#include "client.h"
+#include "AESWrapper.h"
+#include "ResponseUnpacker.h"
+#include "RequestPacker.h"
+
+// --- Client Info ---
+#define CLIENT_VER 3
+#define CLIENT_NAME_MAX_LEN 100
+#define CLIENT_ID_SIZE 16
+#define CLIENT_ID_CELL 3
+#define CLIENT_ID_OFFSET 11
+
+// --- Buffer Sizes ---
+#define FILE_CHUNK_SIZE 1024
+#define BUFFER_SIZE 1024
+
+// --- Port Info ---
+#define DEFAULT_PORT 1256
+
+// --- AES Key ---
+#define AES_KEY_CELL 4
+#define AES_KEY_OFFSET 19
+
+// --- CRC ---
+#define CRC_CELL 6
+#define CRC_OFFSET 5
+#define CRC_SUCCESS_REQUEST_NUM 900
+#define CRC_FAIL_REQUEST_NUM 901
+#define CRC_TOTAL_FAIL_REQUEST_NUM 902
+
+// --- Error Handling ---
+#define PROCESSING_ERROR 2
+#define ERROR 1
+
+// --- Processing Status ---
+#define PROCESSING_SUCCESS 0
+#define PROCESSING_FAIL 1
+
+// --- Registration Status ---
+#define REGISTRATION_ERROR 1
+#define REGISTRATION_SUCCESS 0
+
+// --- Reconnection Status ---
+#define RECONNECTION_SUCCESS 0
+#define RECONNECTION_ERROR 1
+
+// --- Retry Attempts ---
+#define NUM_OF_TRIES_FOR_SERVER_FAIL 4
+
+// --- Other Codes ---
+#define CODE_CELL 1
+#define SUCCESS 0
+
+class FileProcessor {

```

Comment:
missleading name, this is the highest hirearchy class you have and its called file processor.
you should split the actual processing of files to a different class and make this just a manager with ***Businnes Logic***